### PR TITLE
add liquibase script and init persistance layer. add controller for saving orders

### PR DESCRIPTION
**Pull request by andrei-astaschenko**

### CODE REVIEW

File: src/main/java/com/epam/petclinic/order/domain/Order.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package com.epam.petclinic.order.domain;
+
+import javax.persistence.*;
+import java.util.List;
+
+@Entity
+@Table(name = "order", schema = "app")
+public class Order {
+    @Id
+    @Column(name = "uid_order")
```
Comment (Andrei-Khadziukou) [inline src/main/java/com/epam/petclinic/order/domain/Order.java]: Use `order_id`

File: src/main/java/com/epam/petclinic/order/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.Collections;
+
+@RestController
+public class OrderController {
+
+    @Autowired
+    private IOrderService orderService;
+
+    @RequestMapping(name = "/orders", method = RequestMethod.POST)
+    public Collection<String> saveOrder(@RequestBody Order order) {
```
Comment (Andrei-Khadziukou) [inline src/main/java/com/epam/petclinic/order/controller/OrderController.java]: What is the purpose to return collection of strings?

File: src/main/java/com/epam/petclinic/order/domain/Order.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    private String id;
+    @Column(name = "animal_uid")
+    private String animalId;
+    @Transient
+    private List<String> serviceIds;
+
+    public Order() {
+    }
+
+    public Order(String animalId, List<String> serviceIds) {
```
Comment (Andrei-Khadziukou) [inline src/main/java/com/epam/petclinic/order/domain/Order.java]: Is there any usage of this constructor?

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Build folders
```
Comment (Andrei-Khadziukou) [inline .gitignore]: Skip that file

File: src/main/java/com/epam/petclinic/order/controller/OrderController.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Collection;
+import java.util.Collections;
+
+@RestController
+@RequestMapping("/api/orders")
```
Comment (Andrei-Khadziukou) [inline src/main/java/com/epam/petclinic/order/controller/OrderController.java]: remove '/api'

File: README.md [hunk 6]
```diff
@@ -1,4 +1,13 @@
 # petclinic-order
 
+## Liquibase manual configuration
+You need to execute next SQL queries to init PostgreSQL DB:
+- CREATE USER pc_admin;
```
Comment (Andrei-Khadziukou) [inline README.md:5]: you use specific password below... where did you mention that password when creating user?

File: src/main/resources/db/changelog/db.changelog.xml [hunk 7]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet id="1" author="andrei_astaschenko">
+        <createTable tableName="pc_order" schemaName="app">
```
Comment (Andrei-Khadziukou) [inline src/main/resources/db/changelog/db.changelog.xml]: could you please check that automatic rollback is working for a such liquibase script

File: src/main/resources/db/changelog/db.changelog.xml [hunk 8]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet id="1" author="andrei_astaschenko">
```
Comment (Andrei-Khadziukou) [inline src/main/resources/db/changelog/db.changelog.xml:8]: @susau, we should think about naming convention for table names and schemas, as well as about identifiers of changeSet-s.

File: src/main/resources/db/changelog/db.changelog.xml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        <createTable tableName="pc_order" schemaName="app">
+            <column name="pc_order_id" type="varchar(255)">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="pc_name" type="varchar(255)"/>
```
Comment (Andrei-Khadziukou) [inline src/main/resources/db/changelog/db.changelog.xml]: what does `pc_` mean?

Comment (andrei-astaschenko) [reply src/main/resources/db/changelog/db.changelog.xml]: It means 'pet clinic'. I added question about naming convention in PoC documentation.

File: src/main/resources/db/changelog/db.changelog.xml (hunk 8 above)
Comment (andrei-astaschenko) [reply src/main/resources/db/changelog/db.changelog.xml:8]: I added question about naming convention in PoC documentation.

File: build.gradle [hunk 10]
```diff
@@ -31,6 +31,9 @@ dependencies {
     compile('org.liquibase:liquibase-core')
     compile('org.springframework.boot:spring-boot-starter-validation')
     compile('org.springframework.boot:spring-boot-starter-web')
+    compile('org.springframework.boot:spring-boot-starter-data-jpa')
+    compile('org.apache.commons:commons-lang3:3.3.2')
```
Comment (Andrei-Khadziukou) [inline build.gradle:38]: @susau, how to manage version through all applications?

File: README.md (hunk 6 above)
Comment (andrei-astaschenko) [reply README.md:5]: Specified password was removed because of it isn't necessity.

File: build.gradle [hunk 11]
```diff
@@ -7,12 +7,15 @@ buildscript {
     }
     dependencies {
         classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
+        classpath 'org.liquibase:liquibase-gradle-plugin:1.2.4'
```
Comment (Andrei-Khadziukou) [inline build.gradle:10]: @susau, how to manage versions of dependencies?

**Pull request merged**